src/ssl/bio.cc

```diff
@@ -76,7 +76,11 @@ Ssl::Bio::Create(const int fd, Security::Io::Type type)
         BIO_meth_set_create(SquidMethods, squid_bio_create);
         BIO_meth_set_destroy(SquidMethods, squid_bio_destroy);
     }
+#if defined(LIBRESSL_VERSION_NUMBER)
+    BIO_METHOD *useMethod = SquidMethods;
+#elif !defined(LIBRESSL_VERSION_NUMBER)
     const BIO_METHOD *useMethod = SquidMethods;
```

Comment:
How about removing `const` unconditionally, without testing for `LIBRESSL_VERSION_NUMBER`? SquidMethods is not defined to point to `const BIO_METHOD`. If that works for both LibreSSL and OpenSSL, then we should use `auto` instead of `BIO_METHOD *`.

Please do not misinterpret my comment as a sign that Squid officially supports LibreSSL today.

Reply:
My test confirms that auto here will build. I'm not able to test runtime but it should be fine.

@Brightside56, if you are able to confirm that and/or test the runtime works properly please do that auto as the type for both #if and #else code blocks.